boards/shared/Inc/App/App_SharedError.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
```

Comment:
Another small note is that I split things up in two files (`SharedError` and `SharedErrorTable` because I felt like it made things much easier to reason about.